bin/node/src/chain_spec.rs

```diff
@@ -33,6 +34,38 @@ pub fn get_from_seed<TPublic: Public>(seed: &str) -> <TPublic::Pair as Pair>::Pu
 
 type AccountPublic = <Signature as Verify>::Signer;
 
+#[derive(Clone)]
+pub struct SerializablePeerId {
+    inner: PeerId,
+}
+
+impl SerializablePeerId {
+    pub fn new(inner: PeerId) -> SerializablePeerId {
+        SerializablePeerId { inner }
+    }
+}
+
+impl Serialize for SerializablePeerId {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let s: String = format!("{}", self.inner);
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> Deserialize<'de> for SerializablePeerId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        let inner = PeerId::from_str(&s).expect("Could not deserialize as a PeerId");
```

Comment:
We shouldn't panic in  a function returning `Result`. Replace `.expect(...)` with something like 
`.map_err(|_| D::Error::custom(format!("iinvalid PeerId: {}", s)))?` instead?

Reply:
:+1: